### PR TITLE
fix(ci): disable licensei check as it currently hits GitHub API rate limits

### LINKED ISSUE #406: flakey `License Checks` checks
We are getting GitHub api rate limit exceeded

More details can be found in this log file
[pipelinesghubeus13.actions.githubusercontent.com_4S3JTRwu6GPB54RzNDE0wufbZC2P6Ks0uZqRow8Hx060p5Ckak__apis_pipelines_1_runs_1751_signedlogcontent_8_urlExpires=2023-09-14T12%3A56%3A43.1729041Z&urlSigningMethod=HMACV1&urlSignature=A3VOxnSX8uHQhKppDUFHkAf.txt](https://github.com/opentffoundation/opentf/files/12608955/pipelinesghubeus13.actions.githubusercontent.com_4S3JTRwu6GPB54RzNDE0wufbZC2P6Ks0uZqRow8Hx060p5Ckak__apis_pipelines_1_runs_1751_signedlogcontent_8_urlExpires.2023-09-14T12.3A56.3A43.1729041Z.urlSigningMethod.HMACV1.urlSignature.A3VOxnSX8uHQhKppDUFHkAf.txt)

### PR DESCRIPTION
**Pull request by pdecat**
fix #406

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -203,6 +203,7 @@ jobs:
 
       - name: Run licensei
         env:
-           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
+          export GITHUB_TOKEN
```
Comment (eranelbaz) [inline .github/workflows/checks.yml:208]: do we still need that `export`?

Comment (pdecat) [reply .github/workflows/checks.yml:208]: It is was fixes the CI

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -203,6 +203,7 @@ jobs:
 
       - name: Run licensei
         env:
-           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (pdecat) [inline .github/workflows/checks.yml:206]: This space change is purely stylistic.

Comment (pdecat) [reply .github/workflows/checks.yml:206]: Can remove this change if desired.

File: .github/workflows/checks.yml (hunk 1 above)
Comment (pdecat) [reply .github/workflows/checks.yml:208]: Marking the PR draft again to re-verify this...

File: .github/workflows/checks.yml (hunk 2 above)
Comment (pdecat) [reply .github/workflows/checks.yml]: Removed this change to verify https://github.com/opentffoundation/opentf/pull/408#discussion_r1325947154

**Comment by pdecat** [comment]
Hi @eranelbaz, care to approve the GitHub Actions runs?

**Comment by pdecat** [comment]
@eranelbaz Oh :facepalm: You were right, it is the indentation of the `env:` that fixes the issue, sick.

Updating the PR...

**Comment by eranelbaz** [comment]
@pdecat sure, update me when you done
Thank you!

File: .github/workflows/checks.yml (hunk 2 above)
Comment (pdecat) [reply .github/workflows/checks.yml:206]: This is *what* fixes the CI :facepalm: ...

**Comment by pdecat** [comment]
@eranelbaz done!

**Comment by pdecat** [comment]
At least, I've learned something today...

**Review by rumenvasilev — APPROVED** [review]
YAML FTW... :)

**Comment by pdecat** [comment]
Could it be possible that licensei is actually consuming the GITHUB_TOKEN quotas?

**Comment by rumenvasilev** [comment]
> Hmm, it is failing again now 🤔

Are you sure you can use the default GITHUB_TOKEN? IIRC it's restricted to this repository only. So if you try to access other repositories, it might be as good as an anonymous one.

**Comment by rumenvasilev** [comment]
> > Are you sure you can use the default GITHUB_TOKEN? IIRC it's restricted to this repository only. So if you try to access other repositories, it might be as good as an anonymous one.
> 
> Well, it worked in this run https://github.com/opentffoundation/opentf/actions/runs/6185956647/job/16792599028?pr=408

It failed two times actually: https://github.com/opentffoundation/opentf/actions/runs/6185956647/job/16792599028?pr=408#step:4:239

**Comment by rumenvasilev** [comment]
Rate-limit per repo per hour is 1000:
https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limits-for-requests-from-github-actions

**Comment by rumenvasilev** [comment]
> > Rate-limit per repo per hour is 1000: https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limits-for-requests-from-github-actions
> 
> Ok. That's way lower than with a personal token that I've been using locally.
> 
> Looking into modifying licensei to not hit GitHub API if a package is already available in `vendor/`.
> 
> In the mean time, what about disabling this check?

If you make it skip everything that's in vendor, what's the point of licensei? Is there anything preventing anyone from committing vendor data with their PR. In that case, what is going to gate against banned licenses?

**Comment by pdecat** [comment]
> If you make it skip everything that's in vendor, what's the point of licensei?

`licensei` checks both sources for searching `LICENSE` files. I do not intend to make it skip what's in `vendor/`, just not also ask GitHub for it.

> Is there anything preventing anyone from committing vendor data with their PR.

Not yet, `vendor/` is just in `.gitignore` right now.
But that a good  idea for an additional check!

> In that case, what is going to gate against banned licenses?

The license check step runs `go mod vendor` before running licensei.

**Comment by pdecat** [comment]
Submitted #412 to disable license checks in CI until a better solution is found.

**Pull request merged**